File: DaggerMVP/app/src/main/java/com/example/cpu10225/daggermvp/di/component/ApplicationComponent.java
package com.example.cpu10225.daggermvp.di.component;

import android.app.Application;
import android.content.Context;

import com.example.cpu10225.daggermvp.App;
import com.example.cpu10225.daggermvp.data.DataManager;
import com.example.cpu10225.daggermvp.di.module.ApplicationModule;
import com.example.cpu10225.daggermvp.util.anotation.ApplicationContext;

import javax.inject.Singleton;

import dagger.Component;
import retrofit2.Retrofit;

/**
 * Created by cpu10225 on 17/11/2017.
 */

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {
    void inject(App app);
    Application application();
    DataManager dataManager();
    @ApplicationContext
    Context context();
}

File: KotlinOnAndroid/app/src/main/java/com/example/cpu10225/kotlinonandroid/di/component/ApplicationComponent.kt
package com.example.cpu10225.kotlinonandroid.di.component

import android.app.Application
import com.example.cpu10225.kotlinonandroid.MyApp
import com.example.cpu10225.kotlinonandroid.data.DataManager
import com.example.cpu10225.kotlinonandroid.di.module.ApplicationModule
import com.example.cpu10225.kotlinonandroid.util.annotation.CustomScope
import dagger.Component
import javax.inject.Singleton

/**
 * Created by cpu10225 on 05/12/2017.
 */

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {
    fun inject(myApp: MyApp)
    fun application(): Application
    fun dataManager(): DataManager
}